internal/controllers/app_controller.go

```diff
@@ -443,12 +443,14 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App, logger
 				}
 			}
 
-			err = r.watchDeployEvents(ctx, app, &wc, wl, &process, r.Recorder)
+			watcher, err := initWatcher(ctx, app, &wc, wl, &process, r.Recorder)
 			if err != nil {
 				return appReconcileResult{
 					err: fmt.Errorf("failed to get deploy events: %w", err),
 				}
 			}
+			go r.watchDeployEvents(ctx, app, &wc, wl, &process, r.Recorder, watcher)
```

Comment:
we probably want to handle the returned error

Reply:
We create Events for errors inside `watchDeployEvents`. Did you mean something else? If we check the error, it will block the goroutine.